modules/imgproc/test/test_houghlines.cpp

```diff
@@ -340,6 +340,53 @@ TEST(HoughLines, regression_21983)
     EXPECT_NEAR(lines[0][1], 1.57179642, 1e-4);
 }
 
+TEST(WeightedHoughLines, horizontal)
+{
+    Mat img(25, 25, CV_8UC1, Scalar(0));
+    // draw lines. from top to bottom, stronger to weaker.
+    line(img, Point(0, 6), Point(25, 6), Scalar(255));
+    line(img, Point(0, 12), Point(25, 12), Scalar(254));
+    line(img, Point(0, 18), Point(25, 18), Scalar(253));
+    
+    // detect lines
+    std::vector<Vec2f> lines;
+    int threshold{253*25-1};
+    bool use_edgeval{true};
+    HoughLines(img, lines, 1, CV_PI/180, threshold, 0, 0, 0.0, CV_PI, use_edgeval);
+    
+    // check results
+    ASSERT_EQ(lines.size(), 3U);
+    // detected lines is assumed sorted from stronger to weaker. 
+    EXPECT_EQ(lines[0][0], 6);
+    EXPECT_EQ(lines[1][0], 12);
+    EXPECT_EQ(lines[2][0], 18);
```

Comment:
Signature for `_EQ` is `ASSERT_EQ(expected_reference, actual_result);`.
Correct order of parameters are required to emit valid error messages.

Reference: https://github.com/opencv/opencv/blob/4.0.0/modules/ts/include/opencv2/ts/ts_gtest.h#L8196-L8200

```
GTEST_API_ AssertionResult EqFailure(const char* expected_expression,
                                     const char* actual_expression,
                                     const std::string& expected_value,
                                     const std::string& actual_value,
                                     bool ignoring_case);
```

Reply:
I'm sorry. I fixed it.